nilearn/maskers/nifti_labels_masker.py

```diff
@@ -283,9 +283,10 @@ def _sanitize_labels(self, labels):
                     stacklevel=3,
                 )
             if not all(isinstance(x, str) for x in labels):
+                types_labels = {type(x) for x in labels}
```

Comment:
Is this related?

Reply:
Actually yes. Indirectly.

the pyupgrade hook was complaining about this line: https://github.com/nilearn/nilearn/actions/runs/8844898238/job/24287692390#step:5:470

because we can use generator for this

`set([type(x) for x in labels])` --> `{type(x) for x in labels}`

I extracted it out of the f string because a double curly braces is used as an escape character

```python
>>> f"{{type(x) for x in ['foo', 1, 2.0]}}"
"{type(x) for x in ['foo', 1, 2.0]}"

>>> a={type(x) for x in ['foo', 1, 2.0]}
>>> f"{a}"
"{<class 'float'>, <class 'int'>, <class 'str'>}"
```